### PR TITLE
replace deprecated grep stuff

### PR DESCRIPTION
**Pull request by neheb**
the old grep stuff exposes an issue: https://github.com/openwrt/packages/issues/17803

Might as well fix it everywhere.

### CODE REVIEW

**Comment by neheb** [comment]
The first commit fixes the linked issue. Or works around it depending on how you look at it.

**Review by SiNONiMiTY — APPROVED** [review]
Same functionality and removes depreciated dependencies.

File: target/linux/bcm4908/base-files/lib/upgrade/platform.sh [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # SPDX-License-Identifier: GPL-2.0 OR BSD-2-Clause
 
-RAMFS_COPY_BIN="bcm4908img expr egrep fdtget fw_printenv fw_setenv tr"
+RAMFS_COPY_BIN="bcm4908img expr grep fdtget fw_printenv fw_setenv tr"
```
Comment (hauke) [inline target/linux/bcm4908/base-files/lib/upgrade/platform.sh:3]: Why don't you add a `-E` here?

Comment (neheb) [reply target/linux/bcm4908/base-files/lib/upgrade/platform.sh:3]: This is a list of binaries, not commands.

**Comment by hauke** [comment]
Thank you for your patches, I applied them to master in 7994461a5a1c2cdd8bf4cc3fa38c4d496534f080 and 8cfce165a7e3605c22b045cdc1f745b4d6b49fb9.

**Comment by neheb** [comment]
@hauke Assuming the new release of 21.02 isn't cut yet, can you backport the first commit? The linked issue is also a problem with 21.02.

**Pull request closed without merging**